site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+
+This syntax has limited utility currently, but could become more relevant if additional functionality beyond traffic splitting is added to the `HTTPRoute` resource. `HTTPRoutes` configured in this way MAY be used to manually enroll services into a mesh (which could trigger behavior like injecting a sidecar proxy) if they have not already been enrolled by some other mechanism (as proposed in [GEP-1291: Mesh Representation](https://docs.google.com/document/d/1oyA9uUH7pNNxxwy3WZGSWx-edHDBLrujcezr8q3el70/edit) or similar).
+
+```
+metadata:
+  name: foo_route
+  namespace: store
+spec:
+  parentRefs:
+  - kind: Service
+    name: foo
+  rules:
+    matches:
+    - path:
+        value: "/bar"
+```
+
+The example above would drop all incoming traffic for HTTP paths other than `/bar` to the `foo` `Service`, following the existing spec to return HTTP 404 response codes for unmatched requests, and HTTP 500 response codes for requests excluded due to an [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter).
+
+### Allowed `Service` types
+
+Valid `Service` resources to be selected as a `parentRef` SHOULD provision one or more virtual IP(s) and DNS hostname(s) within the Kubernetes cluster, and include the default [`ClusterIP`](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types) `Service` type, with or without selectors.
+
+`Service` resource with [`type: NodePort`](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) or [`type: LoadBalancer`](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) MUST NOT be allowed as `parentRefs` or `backendRefs`. It should isntead be preferred to expose services publicly through the North/South Gateway API interfaces.
+
+`Services` with [`type: ExternalName`](https://kubernetes.io/docs/concepts/services-networking/service/#externalname) SHOULD NOT be allowed as `parentRefs` or `backendRefs` due to [security concerns](https://github.com/kubernetes/kubernetes/issues/103675), although might eventually play some role in [configuring egress functionality](https://github.com/kubernetes-sigs/gateway-api/issues/1070).
+
+`Service` resources supported as `backendRefs` SHOULD be consistent with expectations for North/South Gateway API implementations, MUST have associated endpoints, and include:
+
+* `ClusterIP` `Services` with selectors
+* [Headless Services](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) with selectors. Headless services without selectors SHOULD NOT be supported due to the possibility of matching `Services` with `type: ExternalName`.
+* `StatefulSet` resources
```

Comment:
```suggestion
* `StatefulSet` resources, by targeting the associated headless service
```
@costinm you mentioned this may be difficult for some meshes to support, should we instead remove `StatefulSet` from this section and just keep it as a MAY further down?

Reply:
I agree with this change. Targeting a statefulset specifically sounds like the wrong abstraction

Reply:
Targetting headless service is not un-ambiguous either btw. Its *possible* to implement, but you can have 2 headless services for 1 pod, and its impossible to distinguish them based on IP ( since there is no VIP ). If we want to allow this, lets clearly document the behavior.

Reply:
In Istio I think we just pick the oldest or something. its pretty rare to have this scenario

Reply:
Opened https://github.com/kubernetes-sigs/gateway-api/issues/1484 to continue this discussion.